README.md

```diff
@@ -1 +1,226 @@
 # origin-metrics
+
+TODO: write brief introduction here.
+
+## Building the Docker Containers
+
+The docker containers can be built using the `hack/build-images.sh` script.
+
+	cd hack
+	./build-images.sh --prefix=openshift/origin- --version=devel
+
+Where the `--prefix` value is used to specify a prefix to the docker image and the `--version` value is used to specify the version tag for the resulting docker images.
+
+For instance, as with the above command, a prefix of `openshift/origin-` and version `devel` will create the following docker images:
+
+	openshift/origin-metrics-deployer
+	openshift/origin-metrics-heapster
+	openshift/origin-metrics-cassandra
+	openshift/origin-metrics-hawkular-metrics
+	
+## Setting up your System
+
+The Metrics components require a properly running OpenShift instances. How to properly install and configure OpenShift is beyond the scope of this document. Please see the [OpenShift documentation](https://docs.openshift.org/latest/welcome/index.html) on how to properly install and setup your system.
+
+### Enabling the Read Only Kubelet Endpoint
+
+Currently the kubelet endpoints are secured with certificate authentication and are not accessible to metric contianers. This is a known issue which is currently being worked on: https://github.com/openshift/origin/pull/4873 
+
+Until this is properly fix in OpenShift, you will need to enable the RO endpoint for each of your nodes.
+
+For each of your node's `node-config.yaml` you will need to run the following command to enable read only access:
+
+	cat >> node-config.yaml << DONE
+	kubeletArguments:
+ 	read-only-port:
+ 	- "10266"
+	DONE
+
+### Creating Persistent Storage
+
+The Cassandra database stores its data to persistent storage. For each Cassandra node you deploy, you will need a persistent volume with sufficient data available. You do not need to directly manage peristent volume claims as the deployer and templates will take care of that for you.
+
+Please see the [OpenShift documentation](https://docs.openshift.org/latest/architecture/additional_concepts/storage.html) for how to setup and configure persistent volumes.
+
+For example if you have a NFS server running on localhost with an exposed directory at `/persistent_storage/pv01`, the following command will generate a 10 gigabyte persistent volume:
+
+	oc create -f - <<API
+	apiVersion: v1
+	kind: PersistentVolume
+	metadata:
+	  name: my_pv
+	spec:
+	  capacity:
+	    storage: 10Gi
+	  accessModes:
+	    - ReadWriteOnce
+	    - ReadWriteMany
+	  persistentVolumeReclaimPolicy: Recycle
+	  nfs:
+	    server: localhost
+	    path: /persistent_storage/pv01
+	API
+
+
+## Deploying
+
+
+### Create a Metrics Project
+
+To prevent unwanted components from accessing the metric's persistent volume claim, it is highly recommended to deploy all the metric components within their own project. Management of the metric components is also easier if they are separated in their own project as well.
```

Comment:
...within their own project (e.g. metrics-infa).....

I only say this instead of 'metrics' here because the operations guys talked about us standing up our infrastructure with projects like this, though they could have said 'infra-metrics', I don't quite remember.